### DISCUSSION TITLE
Change zoom, unable to go full screen.

### DISCUSSION THREAD

**Discussion by maxwangwyf** [TomSchimansky/CustomTkinter#2595] [Q&A]
Change zoom, unable to go full screen.
if i change zoom to 125%  ，Full screen is not effective.

**Comment by Akascape** [comment]
Can you provide an example?

**Reply by maxwangwyf** [reply]
root = ctk.CTkToplevel()
    root.title("Full Screen Time Display")

    root.attributes('-fullscreen', True)
    root.grab_set()
    entry = ctk.CTkEntry(root, font=('Arial', 200), justify='center', fg_color="black", bg_color="black", text_color='white',border_color="black")
    entry.pack(fill="both",expand=True,padx=0,pady=0)

    simulate_input(entry)

    root.bind("<Key>", exit_program)
    root.mainloop()
like this if i Change zoom, i cant full screen

**Reply by Akascape** [reply]
@maxwangwyf Did you try this?
```python
root.attributes('-zoomed', True)
```
or 
```python
root.state("zoomed")
```

If you want to go full screen, try the after method:
```python
root.after(300, lambda: root.attributes('-fullscreen', True))
```

**Reply by maxwangwyf** [reply]
def oswatch():
    def create_oval_line(canvas, x1, y1, x2, y2, width, **kwargs):
        dx = x2 - x1
        dy = y2 - y1
        line_length = math.hypot(dx, dy)
        if line_length < width:
            canvas.create_rectangle(x1, y1, x2, y2, **kwargs)
            return None
        ux = dx / line_length
        uy = dy / line_length
        x_offset = (width / 2) * uy
        y_offset = (width / 2) * -ux
        p1 = (x1 + x_offset, y1 + y_offset)
        p2 = (x2 + x_offset, y2 + y_offset)
        p3 = (x2 - x_offset, y2 - y_offset)
        p4 = (x1 - x_offset, y1 - y_offset)
        re = canvas.create_polygon(p1, p2, p3, p4, **kwargs)
        cr = (width - width / 60) / 2
        c1 = canvas.create_oval(x1 - cr, y1 - cr, x1 + cr, y1 + cr, outline='', **kwargs)
        c2 = canvas.create_oval(x2 - cr, y2 - cr, x2 + cr, y2 + cr, outline='', **kwargs)
        return re, c1, c2

    def draw_ticks(canvas, center_x, center_y, radius_outer, radius_inner):
        for i in range(60):
            angle = i * (math.pi / 30)  # 每个刻度 6 度
            x1 = center_x + radius_outer * math.sin(angle)  # 修改为 sin
            y1 = center_y - radius_outer * math.cos(angle)  # 修改为 -cos
            x2 = center_x + radius_inner * math.sin(angle)  # 修改为 sin
            y2 = center_y - radius_inner * math.cos(angle)  # 修改为 -cos
            width = 8 if i % 5 == 0 else 3  # 每5个刻度绘制长一点的刻度
            canvas.create_line(x1, y1, x2, y2, fill="#EFEFEF", width=width)

    def update_clock(canvas, center_x, center_y, hands):
        canvas.delete("hand")  # 清除时针、分针和秒针

        # 获取当前时间
        now = datetime.now()
        seconds = now.second
        minutes = now.minute
        hours = now.hour % 12  # 将小时转换为12小时制

        # 计算时针的位置
        hour_angle = (hours + minutes / 60) * (math.pi / 6)  # 每小时 30 度
        x_hour_end = center_x + hands[2] * math.sin(hour_angle)  # 修改为 sin
        y_hour_end = center_y - hands[2] * math.cos(hour_angle)  # 修改为 -cos
        create_oval_line(canvas, center_x, center_y, x_hour_end, y_hour_end, 10, fill='#EFEFEF', tags="hand")

        # 计算分针的位置
        minute_angle = (minutes + seconds / 60) * (math.pi / 30)  # 每分钟 6 度
        x_min_end = center_x + hands[1] * math.sin(minute_angle)  # 修改为 sin
        y_min_end = center_y - hands[1] * math.cos(minute_angle)  # 修改为 -cos
        create_oval_line(canvas, center_x, center_y, x_min_end, y_min_end, 6, fill='#EFEFEF', tags="hand")

        # 计算秒针的位置
        second_angle = seconds * (math.pi / 30)  # 每秒 6 度
        x_sec_end = center_x + hands[0] * math.sin(second_angle)  # 修改为 sin
        y_sec_end = center_y - hands[0] * math.cos(second_angle)  # 修改为 -cos
        create_oval_line(canvas, center_x, center_y, x_sec_end, y_sec_end, 4, fill='#E45D47', tags="hand")

        small_circle_radius = 10
        canvas.create_oval(center_x - small_circle_radius, center_y - small_circle_radius,
                        center_x + small_circle_radius, center_y + small_circle_radius,
                        fill='#E45D47', outline='')

        # 更新时间
        canvas.after(1000, update_clock, canvas, center_x, center_y, hands)

    def exit_program(event):
        root.destroy()  # 关闭窗口

    # 创建主窗口
    root = ctk.CTkToplevel()
    root.title("Analog Clock")
    
    # 设置全屏
    root.attributes('-fullscreen', True)
    
    root.grab_set()
    # 获取屏幕尺寸
    screen_width = root.winfo_screenwidth()
    screen_height = root.winfo_screenheight()

    # 创建画布并居中
    canvas = ctk.CTkCanvas(root, width=screen_width, height=screen_height, bg='#2B2A2F', highlightbackground='#2B2A2F', highlightcolor='#2B2A2F')
    canvas.pack()

    # 时钟中心坐标
    center_x = screen_width // 2
    center_y = screen_height // 2

    # 定义半径和指针长度
    hand = [220, 200, 140]  # [秒针长度, 分针长度, 时针长度]

    # 绘制刻度
    draw_ticks(canvas, center_x, center_y, 250, 230)

    # 启动时钟动画，并根据当前时间初始化
    update_clock(canvas, center_x, center_y, hand)
    root.bind("<Key>", exit_program)
    root.mainloop()
the window fullscreen,but the frame not fullscreen
